src/abstractitensornetwork.jl

```diff
@@ -798,7 +798,11 @@ function hascommoninds(
   ::typeof(siteinds), A::AbstractITensorNetwork{V}, B::AbstractITensorNetwork{V}
 ) where {V}
   for v in vertices(A)
-    !hascommoninds(siteinds(A, v), siteinds(B, v)) && return false
+    if isempty(siteinds(A, v))
+      !isempty(siteinds(B, v)) && return false
+    else
+      !hascommoninds(siteinds(A, v), siteinds(B, v)) && return false
+    end
```

Comment:
I'm confused by this change. It seems like this catches the case when `A` doesn't have site indices on vertex `v`, but `B` has site indices on vertex `v`. That seems like a case that `hascommoninds` should be working for. What goes wrong in `hascommoninds` in that case?

Reply:
When `A` doesn't have site indices on vertex `v`, but `B` has site indices on vertex `v`, it will return `false`, while it will continue the loop if both don't have site indices on vertex `v`. The `false` would indeed also be caught in the `else` clause, so it would indeed be more transparent to write

Reply:
Gotcha, thanks. I see that is a strange corner case. See my comment https://github.com/mtfishman/ITensorNetworks.jl/pull/144#issuecomment-1983581282, I think it would be simpler to just remove these functions entirely, since they are likely doing more harm than good, and we can reassess what kind of checks we really want in this functions, if any.